src/smolagents/agents.py

```diff
@@ -1171,22 +1168,26 @@ def __init__(
                 "Caution: you set an authorization for all imports, meaning your agent can decide to import any package it deems necessary. This might raise issues if the package is not installed in your environment.",
                 0,
             )
-
-        if use_e2b_executor and len(self.managed_agents) > 0:
-            raise Exception(
-                f"You passed both {use_e2b_executor=} and some managed agents. Managed agents is not yet supported with remote code execution."
-            )
-
-        if use_e2b_executor:
-            self.python_executor = E2BExecutor(
-                self.additional_authorized_imports,
-                self.logger,
-            )
-        else:
-            self.python_executor = LocalPythonInterpreter(
-                self.additional_authorized_imports,
-                max_print_outputs_length=max_print_outputs_length,
-            )
+        self.executor_type = executor_type
+        self.executor_kwargs = executor_kwargs or {}
+        self.python_executor = self.create_python_executor(executor_type, self.executor_kwargs)
+
+    def create_python_executor(self, executor_type: str, kwargs: Dict[str, Any]) -> PythonExecutor:
+        match executor_type:
+            case "e2b" | "docker":
+                if self.managed_agents:
+                    raise Exception("Managed agents are not yet supported with remote code execution.")
+                if executor_type == "e2b":
+                    return E2BExecutor(self.additional_authorized_imports, self.logger, **kwargs)
+                else:
+                    return DockerExecutor(self.additional_authorized_imports, self.logger, **kwargs)
+            case "local":
+                return LocalPythonExecutor(
+                    self.additional_authorized_imports,
+                    max_print_outputs_length=self.max_print_outputs_length,
+                )
+            case _:  # if applicable
+                raise ValueError(f"Unsupported executor type: {executor_type}")
```

Comment:
Then if you decide to go with an enum:
```suggestion
    def create_python_executor(self, executor_type: ExecutorType, kwargs: Dict[str, Any]) -> PythonExecutor:
        match executor_type:
            case ExecutorType.E2B | ExecutorType.DOCKER:
                if self.managed_agents:
                    raise Exception("Managed agents are not yet supported with remote code execution.")
                if executor_type == ExecutorType.E2B:
                    return E2BExecutor(self.additional_authorized_imports, self.logger, **kwargs)
                else:
                    return DockerExecutor(self.additional_authorized_imports, self.logger, **kwargs)
            case ExecutorType.LOCAL:
                return LocalPythonExecutor(
                    self.additional_authorized_imports,
                    max_print_outputs_length=self.max_print_outputs_length,
                )
            case _: 
                raise ValueError(f"Unsupported executor type: {executor_type}")
```

Reply:
Since the cases are still few, I think we can keep with strings for now, and maybe add enum later!